### PR TITLE
add micromamba usage

### PR DESCRIPTION
**Pull request by munali**

### CODE REVIEW

File: conda_lock/conda_lock.py [hunk 1]
```diff
@@ -372,23 +372,31 @@ def aggregate_lock_specs(lock_specs: List[LockSpecification]) -> LockSpecificati
...
-        # micromamba doesn't support --override-channels
-        micromamba=False,
+        mamba=mamba,
+        micromamba=micromamba,
         conda=True,
         conda_exe=True,
     )
+
+    if micromamba and "MAMBA_ROOT_PREFIX" not in os.environ:
+        os.environ["MAMBA_ROOT_PREFIX"] = str(pathlib.Path(_conda_exe) / "mamba_root")
```
Comment (mariusvniekerk) [inline conda_lock/conda_lock.py:391]:
```suggestion
        os.environ["MAMBA_ROOT_PREFIX"] = str(pathlib.Path(_conda_exe).parent / "mamba_root")
```

File: conda_lock/conda_lock.py [hunk 2]
```diff
@@ -372,23 +372,31 @@ def aggregate_lock_specs(lock_specs: List[LockSpecification]) -> LockSpecificati
...
     if conda_executable:
         if pathlib.Path(conda_executable).exists():
             yield conda_executable
         yield shutil.which(conda_executable)
+
     _conda_exe = ensureconda.ensureconda(
-        mamba=not no_mamba,
-        # micromamba doesn't support --override-channels
-        micromamba=False,
+        mamba=mamba,
```
Comment (mariusvniekerk) [inline conda_lock/conda_lock.py:384]: do we have a test that runs a lock using micromamba?

File: conda_lock/conda_lock.py (hunk 1 above)
Comment (munali) [reply conda_lock/conda_lock.py:391]: good catch!

File: conda_lock/conda_lock.py (hunk 2 above)
Comment (munali) [reply conda_lock/conda_lock.py:384]: no, I will add a test.

**Comment by mariusvniekerk** [comment]
Nice.   Once micromamba support CONDA_SUBDIR we can remove the xfail and see about getting it enabled by default.

**Pull request merged**